Broken testcase test/test_host-network-mode/test_host-network-mode.py

Hi,

When running the nginx-proxy test suite on my box the test `test/test_host-network-mode/test_host-network-mode.py` fails with:
```
_____________________________________________________________________________________ test_forwards_to_host_network_container _____________________________________________________________________________________

docker_compose = <docker.client.DockerClient object at 0x7f27a01369d0>, nginxproxy = <conftest.requests_for_docker object at 0x7f279f5c23a0>

    def test_forwards_to_host_network_container(docker_compose, nginxproxy):
        r = nginxproxy.get("http://host-network.nginx-proxy.tld/port")
>       assert r.status_code == 200
E       assert 502 == 200
E        +  where 502 = <Response [502]>.status_code

test/test_host-network-mode/test_host-network-mode.py:12: AssertionError
```

This failure seems related to commit b88d33d where an internal network was added to the test case. Unfortunately the generated `upstream` block for the `host_network` service uses the `net1` internal network which cannot reach the related container from the nginx-proxy instance:
```nginx
# host-network.nginx-proxy.tld/
upstream host-network.nginx-proxy.tld {
    # Container: test_host-network-mode_host-network_1
    #     networks:
    #         container is in host network mode, using test_host-network-mode_net1 gateway IP
    #     IP address: 192.168.96.1
    #     exposed ports: (none)
    #     default port: 80
    #     using port: 8080
    server 192.168.96.1:8080;
}
```

The related code snippet in `nginx.tmpl` is:
```
            {{- range sortObjectsByKeysAsc $.globals.CurrentContainer.Networks "Name" }}
                {{- if and . .Gateway }}
    #         container is in host network mode, using {{ .Name }} gateway IP
                    {{- $ip = .Gateway }}
                    {{- break }}
                {{- end }}
            {{- end }}
```

The problem is that the `.Gateway` property is defined for internal networks as well:
```json
$ docker inspect test_host-network-mode_net1
[
    {
        "Name": "test_host-network-mode_net1",
        "Id": "9ae91347c63ced795b9fa7aa373cb500dabbb234b05a2153aa892bb092a9297a",
        "Created": "2023-09-30T16:17:41.843953935+02:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "192.168.96.0/20",
                    "Gateway": "192.168.96.1"
                }
            ]
        },
        "Internal": true,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "ec10ace56b87183bbf99ee53f38d8768fef7c28b5ab13bf170e4a3d24f537635": {
                "Name": "test_host-network-mode_sut_1",
                "EndpointID": "064135f9d213d6c1921a90f01d18173a332eaa403a930bbf8751229c47e72002",
                "MacAddress": "02:42:c0:a8:60:02",
                "IPv4Address": "192.168.96.2/20",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "net1",
            "com.docker.compose.project": "test_host-network-mode",
            "com.docker.compose.version": "1.29.2"
        }
    }
]
```

Instead I think that docker-gen should provide the `.Internal` network property so that `nginx.tmpl` could select a gateway where `.Internal` is false. Then the code snippet above would be replaced with:
```
            {{- range sortObjectsByKeysAsc $.globals.CurrentContainer.Networks "Name" }}
                {{- if and . .Gateway (not .Internal) }}
    #         container is in host network mode, using {{ .Name }} gateway IP
                    {{- $ip = .Gateway }}
                    {{- break }}
                {{- end }}
            {{- end }}
```

Docker release is 20.10.25 from current Debian testing.

--- a/nginx.tmpl
+++ b/nginx.tmpl
@@ -75,7 +75,7 @@
                 {{- continue }}
             {{- end }}
             {{- range sortObjectsByKeysAsc $.globals.CurrentContainer.Networks "Name" }}
-                {{- if and . .Gateway }}
+                {{- if and . .Gateway (not .Internal) }}
     #         container is in host network mode, using {{ .Name }} gateway IP
                     {{- $ip = .Gateway }}
                     {{- break }}

